src/components/Divider.tsx

```diff
@@ -1,70 +1,136 @@
-import React, { useEffect } from 'react'
+import React, { useCallback, useRef, useState } from 'react'
 import { useDispatch } from 'react-redux'
-import { useSelector } from 'react-redux'
 import { css } from '../../styled-system/css'
 import { SystemStyleObject } from '../../styled-system/types'
 import Path from '../@types/Path'
+import State from '../@types/State'
+import ThoughtId from '../@types/ThoughtId'
 import { setCursorActionCreator as setCursor } from '../actions/setCursor'
-import { DIVIDER_MIN_WIDTH, DIVIDER_PLUS_PX } from '../constants'
+import { DIVIDER_MIN_WIDTH } from '../constants'
+import useSelectorEffect from '../hooks/useSelectorEffect'
 import attributeEquals from '../selectors/attributeEquals'
-import rootedParentOf from '../selectors/rootedParentOf'
+import { getAllChildrenAsThoughts } from '../selectors/getChildren'
 import fastClick from '../util/fastClick'
 import head from '../util/head'
+import isDivider from '../util/isDivider'
+import parentOf from '../util/parentOf'
+
+/** Helper function to get widths of thoughts by querying the DOM. */
+const getThoughtWidths = (thoughts: { id: ThoughtId }[]) => {
+  return thoughts.map(thought => {
+    const innerThoughtElement = document.querySelector(
+      `[aria-label="tree-node"][data-id="${thought.id}"] [aria-label="thought"]`,
+    )
+    return innerThoughtElement ? innerThoughtElement.getBoundingClientRect().width : 0
```

Comment:
```suggestion
    return innerThoughtElement?.getBoundingClientRect().width ?? 0
```

This line is fine as-is, but I'll make this suggestion as personal preference for slightly better readability.